### ISSUE TITLE
go guru implement use an invalid scope when GOPATH is opened directly

### ISSUE THREAD

**Issue by leaxoy** [Microsoft/vscode-go#1554]
Here is error message:
ERR Command failed: /Users/leaxoy/go/bin/guru -scope _/Users/leaxoy/go/... -json implements /Users/leaxoy/go/src/test/main.go:#56
guru: no packages specified for pointer analysis scope
: Error: Command failed: /Users/leaxoy/go/bin/guru -scope _/Users/leaxoy/go/... -json implements /Users/leaxoy/go/src/test/main.go:#56
guru: no packages specified for pointer analysis scope

On my mac, I set ${HOME}/go as GOPATH, then I open GOPATH with command `code-insiders ~/go`,  and then press `command+shift+F12` to find interface implementations, this error occurred.
`guru` arg `scope` expect `test/...`, but got `_/Users/leaxoy/go/...`.

**Comment by leaxoy** [comment]
```
dokigo λ go list -e -json ..
{
        "Dir": "/Users/leaxoy/go/src",
        "ImportPath": "_/Users/leaxoy/go/src",
        "Stale": true,
        "StaleReason": "build ID mismatch",
        "Incomplete": true,
        "Error": {
                "ImportStack": [
                        ".."
                ],
                "Pos": "",
                "Err": "no Go files in /Users/leaxoy/go/src"
        }
}
dokigo λ go list -e -json ../..
{
        "Dir": "/Users/leaxoy/go",
        "ImportPath": "_/Users/leaxoy/go",
        "Stale": true,
        "StaleReason": "build ID mismatch",
        "Incomplete": true,
        "Error": {
                "ImportStack": [
                        "../.."
                ],
                "Pos": "",
                "Err": "no Go files in /Users/leaxoy/go"
        }
}
dokigo λ cd chatserver/
chatserver λ go list -e -json
{
        "Dir": "/Users/leaxoy/go/src/dokigo/chatserver",
        "ImportPath": "dokigo/chatserver",
        "Root": "/Users/leaxoy/go",
        "Stale": true,
        "StaleReason": "build ID mismatch",
        "Incomplete": true,
        "Error": {
                "ImportStack": [
                        "dokigo/chatserver"
                ],
                "Pos": "",
                "Err": "no Go files in /Users/leaxoy/go/src/dokigo/chatserver"
        }
}```
if not `Root` field present, , may be we should re-populate scope string.

**Comment by ramya-rao-a** [comment]
Thanks for the idea of looking at the `Root`. I was working on #1545 which is the same as this issue. I was about to add a check where we skip the scope if it starts with `_`. But checking for Root is a better idea.

**Comment by ramya-rao-a** [comment]
Fixed. Need some help testing though. All you need to do is
- Download https://github.com/Microsoft/vscode-go/blob/master/Go-latest.vsix
- Run `code --install-extension Go-latest.vsix` 
- If the above fails with `Error: end of central directory record signature not found`, then clone this repo and use the `Go-latest.vsix` file from the cloned repo
- Reload VS Code

**Comment by ramya-rao-a** [comment]
This fix is now available in the latest update (0.6.78) to the Go extension
